lnrpc/rpc.proto

```diff
@@ -2407,6 +2407,35 @@ message Payment {
 
     ///  The fee paid for this payment in milli-satoshis
     int64 fee_msat = 12 [json_name = "fee_msat"];
+
+    /// The time in UNIX nanoseconds at which the payment was created.
+    int64 creation_time_ns = 13 [json_name = "creation_time_ns"];
+
+    /// The HTLCs made in attempt to settle the payment [EXPERIMENTAL].
+    repeated HTLCAttempt htlcs = 14 [json_name = "htlcs"];
+}
+
+message HTLCAttempt {
+        enum HTLCStatus {
+                IN_FLIGHT = 0;
+                SUCCEEDED = 1;
+                FAILED = 2;
+        }
+
+        /// The status of the HTLC.
+        HTLCStatus status = 1 [json_name = "status"];
+
+        /// The route taken by this HTLC.
+        Route route = 2 [json_name = "route"];
+
+        /// The time in UNIX nanonseconds at which this HTLC was sent.
+        int64 attempt_time_ns = 3 [json_name = "attempt_time_ns"];
+
+        /**
+        The time in UNIX nanonseconds at which this HTLC was settled or failed.
+        This value will not be set if the HTLC is still IN_FLIGHT.
+        */
+        int64 resolve_time_ns = 4 [json_name = "resolve_time_ns"];
```

Comment:
Should we update the main rpc at all? We encourage switching to `routerrpc` because it is safe. Only adding the fields there creates another reason to switch over.

Reply:
what would they switch to? we only have one `listpayments` rpc

Reply:
Maybe this is the time to copy it to the `routerrpc` ?

Reply:
and have two rpcs?

Reply:
We need two at some point. The old one deprecated and the new one already available. Identical to the current SendPayment/SendToRoute situation.

Reply:
what specifically about the current one can't be changed or fixed?

Reply:
Discussed offline. As there is virtual no cost to adding it to the main rpc, we can just do it. Need to think about when to mainline `routerrpc`. 0.9 seems too early